src/java.base/share/classes/sun/security/provider/certpath/X509CertPath.java

```diff
@@ -228,7 +227,7 @@ public X509CertPath(InputStream is, String encoding)
             if (is.markSupported() == false) {
                 // Copy the entire input stream into an InputStream that does
                 // support mark
-                is = new ByteArrayInputStream(readAllBytes(is));
+                is = new ByteArrayInputStream(is.readAllBytes());
```

Comment:
I don't understand why the check for #markSupported is done there. The InputStream constructor of PKCS7 creates a DataInputStream on the InputStream only to then call #readFully. I can't find a place where a call to #mark happens. Since the InputStream constructor reads all bytes anyway I wonder whether we could remove this if and unconditionally do:

PKCS7 pkcs7 = new PKCS7(is.readAllBytes());

Reply:
Good idea. Will improve.
By the way, code in `sun.security.pkcs.PKCS7#PKCS7(java.io.InputStream)`  looks suspicious: it reads only `InputStream.available()` bytes, which doesn't make much sense to me.